src/App.js

```diff
@@ -14,11 +17,122 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.PureComponent {
+  state = {
+    clickedStart: false,
+    goods: goodsFromServer,
+    selectValue: 1,
+  };
+
+  startHideButton = () => {
+    this.setState(state => ({
+      clickedStart: !state.clickedStart,
+    }));
+  };
+
+  reverseGoods = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  };
+
+  sortGoodsAlphabetically = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort(),
+    }));
+  };
+
+  resetGoods = () => {
+    this.setState({
+      goods: goodsFromServer,
+      selectValue: 1,
+    });
+  };
+
+  sortGoodsByLength = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  handleSelect = (event) => {
+    const minLength = event.target.value;
```

Comment:
рекомендую использовать деструктуризацию
```suggestion
    const {value} = event.target;
```